ccmlib/node.py

```diff
@@ -1233,7 +1233,8 @@ def stress_process(self, stress_options, **kwargs):
                 # cassandra-stress has version command
                 stress_options = [o.replace('limit=', 'throttle=') for o in stress_options]
 
-        if not ('-node' in stress_options or '-d' in stress_options):
+        if not ('-node' in stress_options or '-d' in stress_options) and not \
+            any(opt.startswith('-cloudconf')  for opt in stress_options):
```

Comment:
I might suggest this:
```python
@staticmethod
def options_implying_node(stress_options: list) -> bool:
    implying_nodes_parameters = ['-node',  '-d', '-cloudconf']
    
    for opt in stress_options:
         return any(opt.startswith(param) for param in implying_nodes_parameters)
    
def stress_process(self, stress_options, **kwargs):
    if not options_implying_node(stress_options):
        # need to specify address
```

but anyhow adding helper for only one usage... (and it wouldn't have any other usages...)

Reply:
in this case, the purpose of introducing a dedicated helper method is accuracy and readability. not for reusing. the `startswith()` call is not accurate, IMHO.

Reply:
> right I've forgot c-s argument are not standard like using one dash for one digit, and two dash for a full word.
> 
> BTW the options can also be `['-d 10.0.0.1',]` and equal won't accurate.

no, it can't. i am copy-n-paste my ipython session here:
```
In [6]: args = ['ls', '-l /tmp']

In [7]: subprocess.Popen(args)
ls: invalid option -- ' '
Try 'ls --help' for more information.
Out[7]: <Popen: returncode: None args: ['ls', '-l /tmp']>
```
> 
> we can define that `--cloudconf` should come on it's own and align them


> I think it's my mistake that I've used it like this:
> 
> ```python
>         output = node1.stress(['write', 'duration=5s', "no-warmup", '-rate',
>                               'threads=2', f'-cloudconf file={self.config_data_yaml}'])
> ```

yeah, for the same reason, one cannot pass `"-foo bar"` as a single command line argument to `subprocess.Popen(args)` as a single element in the `arg` list. we might want to split it into two, like
```py
  output = node1.stress(['write', 'duration=5s', "no-warmup", '-rate',
                              'threads=2', '-cloudconf`, f`file={self.config_data_yaml}'])
```
so in that case, we can simplify `stress_process()` without the helper like
```py
def stress_process(self, stress_options, **kwargs):
    options_implying_node = ['-d', '-node', '-coudconf']
    if any(opt in options_implying_node for opt in stress_options):
        # no need to specify address
        pass
    else:
        # ...
```
note, instead of using `not`. i choose to add an empty branch with comment just for documenting the purpose of this block.

Reply:
change to match the needed logic